### PR TITLE
Add new test verify_stats_removal (RIAK-2312)

### PR DESCRIPTION
**Pull request by JeetKunDoug**
When a vnode starts up, it creates several vnode-specific stats. When that vnode shuts down (generally due to handoff) those stats were not correctly removed. This test confirms that in the happy-path shutdown case those stats are, in fact, removed. Future work in the next release will handle the vnode crashing as well.

This test verifies the fix in basho/riak_kv#1282

### CODE REVIEW

File: tests/verify_stats_removal.erl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    pass.
+
+get_stats_count_for_non_running_vnodes(Node1) ->
+    spawn_link(Node1, verify_stats_removal, get_stats_remote, [self()]),
+    receive
+        {stats, NumStats} ->
+            NumStats;
+        _Other ->
+            throw("Failed to retrieve count of stats from vnode.")
+    end.
```
Comment (jonmeredith) [inline tests/verify_stats_removal.erl]: May be worth a longish timeout (60s) rather than hang r_t runs.

**Comment by JeetKunDoug** [comment]
Added timeout - test should be ready to go. Confirms that stats are leaking on non-fixed code, not leaking on fixed.

**Comment by mrallen1** [comment]
:+1: 6e63512

**Comment by mrallen1** [comment]
Passes against riak_kv 2.0 08edc1
Fails against riak_kv 2.0 290b35

**Pull request merged**